.circleci/soltest_all.sh

```diff
@@ -32,38 +32,35 @@ EVM_VALUES=(homestead byzantium constantinople petersburg istanbul)
 OPTIMIZE_VALUES=(0 1)
 STEPS=$(( 2 + ${#EVM_VALUES[@]} * ${#OPTIMIZE_VALUES[@]} ))
 
-if (( $CIRCLE_NODE_TOTAL )) && (( $CIRCLE_NODE_TOTAL > 1 ))
+if (( CIRCLE_NODE_TOTAL )) && (( CIRCLE_NODE_TOTAL > 1 ))
 then
     # Run step 1 as the only step on the first executor
     # and evenly distribute the other steps among
     # the other executors.
     # The first step takes much longer than the other steps.
-    if (( $CIRCLE_NODE_INDEX == 0 ))
+    if (( CIRCLE_NODE_INDEX == 0 ))
     then
         RUN_STEPS="1"
     else
-        export CIRCLE_NODE_INDEX=$(($CIRCLE_NODE_INDEX - 1))
-        export CIRCLE_NODE_TOTAL=$(($CIRCLE_NODE_TOTAL - 1))
-        RUN_STEPS=$(seq 2 "$STEPS" | circleci tests split)
+        export CIRCLE_NODE_INDEX=$((CIRCLE_NODE_INDEX - 1))
+        export CIRCLE_NODE_TOTAL=$((CIRCLE_NODE_TOTAL - 1))
+        RUN_STEPS=$(seq 2 "$STEPS" | circleci tests split | xargs)
     fi
 else
-    RUN_STEPS=$(seq "$STEPS")
+    RUN_STEPS=$(seq "$STEPS" | xargs)
 fi
 
-# turn newlines into spaces
-RUN_STEPS=$(echo $RUN_STEPS)
-
```

Comment:
So we ended up with https://github.com/ethereum/solidity/pull/9421#discussion_r455112574 anyway...